src/tests/unit/platform/test_alertsv7_v6_compatibility.py

```diff
@@ -0,0 +1,257 @@
+    "remediation",
+    "threat_cause_actor_name",  # on CB Analytics, the record is truncated on v6 so will not match
+    "process_name"  # just the file name on v6, full path on v7
+}
+
+# Fields on the v6 base or common alert object that do not have an equivalent in v7
+BASE_FIELDS_V6 = {
+    "group_details",
+    "category",
+    "threat_activity_c2",
+    "threat_cause_threat_category",
+    "alert_classification",
+    "threat_cause_actor_process_pid"
+}
+
+# Fields on the v6 CB Analytics alert object that do not have an equivalent in v7
+CB_ANALYTICS_FIELDS_V6 = {
+    "blocked_threat_category",
+    "kill_chain_status",
+    "not_blocked_threat_category",
+    "threat_activity_c2",
+    "threat_activity_dlp",
+    "threat_activity_phish",
+    "threat_cause_vector"
+}
+
+# Fields on the v6 Device Control alert object that do not have an equivalent in v7
+DEVICE_CONTROL_FIELDS_V6 = {"threat_cause_vector"}
+
+# Fields on the v6 Container Runtime alert object that do not have an equivalent in v7
+CONTAINER_RUNTIME_FIELDS_V6 = {
+    "workload_id",
+    "target_value"
+}
+
+# Fields on the v6 Watchlist alert object that do not have an equivalent in v7
+WATCHLIST_FIELDS_V6 = {
+    "count",
+    "document_guid",
+    "threat_cause_vector",
+    "threat_indicators"
+}
+
+# Aggregate all the alert type fields
+ALL_FIELDS_V6 = (CB_ANALYTICS_FIELDS_V6 | BASE_FIELDS_V6 | DEVICE_CONTROL_FIELDS_V6 | WATCHLIST_FIELDS_V6
+                 | CONTAINER_RUNTIME_FIELDS_V6 | SKIP_FIELDS)
+
+
+@pytest.mark.parametrize("url, v7_api_response, v6_sdk_response", [
+    ("/api/alerts/v7/orgs/test/alerts/6f1173f5-f921-8e11-2160-edf42b799333", GET_ALERT_v7_CB_ANALYTICS_RESPONSE,
+     ALERT_V6_INFO_CB_ANALYTICS_SDK_1_4_3),
+    ("/api/alerts/v7/orgs/test/alerts/f6af290d-6a7f-461c-a8af-cf0d24311105", GET_ALERT_v7_WATCHLIST_RESPONSE,
+     ALERT_V6_INFO_WATCHLIST_SDK_1_4_3),
+    ("/api/alerts/v7/orgs/test/alerts/46b419c8-3d67-ead8-dbf1-9d8417610fac", GET_ALERT_v7_CONTAINER_RUNTIME_RESPONSE,
+     ALERT_V6_INFO_CONTAINER_RUNTIME_SDK_1_4_3),
+    ("/api/alerts/v7/orgs/test/alerts/2be0652f-20bc-3311-9ded-8b873e28d830", GET_ALERT_v7_HBFW_RESPONSE,
+     ALERT_V6_INFO_HBFW_SDK_1_4_3),
+    ("/api/alerts/v7/orgs/test/alerts/b6a7e48b-1d14-11ee-a9e0-888888888788", GET_ALERT_v7_DEVICE_CONTROL_RESPONSE,
+     ALERT_V6_INFO_DEVICE_CONTROL_SDK_1_4_3)
+])
+def test_v7_generate_v6_json(cbcsdk_mock, url, v7_api_response, v6_sdk_response):
+    """
+    Test the generation of a v6 to_json output
+
+    Compare what is generated by the current SDK with expected from SDK 1.4.3
+    Parameterization above is used to call this test multiple times to test different alert types
+    """
+    # set up the mock request and execute the mock v7 API call
+    cbcsdk_mock.mock_request("GET", url, v7_api_response)
+    api = cbcsdk_mock.api
+    alert = api.select(BaseAlert, v6_sdk_response.get("id"))
+    print("\n\nStarting comparison of {} type alerts".format(alert.get("type")))
```

Comment:
We don't tend to have print statements in tests to reduce the noise log statements are fine as they get suppressed especially when passing

Reply:
removed